mountpoint-s3/src/fs.rs

```diff
@@ -329,9 +329,12 @@ where
             size
         );
 
-        let file_handles = self.file_handles.read().await;
-        let Some(handle) = file_handles.get(&fh) else {
-            return reply.error(libc::EBADF);
+        let handle = {
```

Comment:
How is this different from previous implementation?

Reply:
With the previous implementation, `file_handles` was in the scope of the whole `read` function. Only at the end of the function would it be dropped, and the read lock released.

With this new code, we introduce a new scope (due to the use of `{ ... }`). We acquire the read lock, grab a reference to the file handle we want and return it from the block denoted by `{ ... }`. In doing so, we drop `file_handles` inside the block, and only the cloned file handle itself is kept in scope.

Reply:
We're moving the file handles locking (`self.file_handles.read()`) into a new block, which mean as soon as this block is out of scope the lock will also be release.